pkl-core/src/main/java/org/pkl/core/EvaluatorImpl.java

```diff
@@ -366,24 +377,30 @@ private void handleTimeout(@Nullable TimeoutTask timeoutTask) {
   }
 
   private VmException moduleOutputValueTypeMismatch(
-      VmTyped module, PClassInfo<?> expectedClassInfo, Object value, VmTyped output) {
+      VmTyped module,
+      PClassInfo<?> expectedClassInfo,
+      Object value,
+      VmTyped container,
+      Identifier propertyIdentifier,
+      String propertyName) {
```

Comment:
There's a fair bit of logic here that's bespoke to reading `output.value`, so it's not too appropriate to be re-purposed. For example, it show different error messages based on whether `output.value` was explicitly assigned or not.

I think we should just copy some of the logic here that's useful, and inline that into `readModuleOutput`.

Reply:
The amend hint is for things like: your module does not amend anything, and maybe it should have `amends "pkl:Project"`.

In this case, `output` should always be of type `ModuleOutput`, so that type of suggestion doesn't really make much sense.

Showing the source section of the assignment makes sense; I think you can either copy/paste it, or maybe refactor this into a separate method?